File: frontend/src/services/bookService.js
import http from "../http-common";

class BookService{
    getAllBook(){
        return http.get("/");
    }
    getBookInfoById(id){
        return http.get(`/${id}`);
    }
    saveBookInfo(data){
        return http.post("/save",data);
    }
    updateBookInfo(id,data){
        return http.put(`/${id}`,data);
    }
    deleteBookInfoById(id){
        return http.delete(`/${id}`);
    }
    
}

export default new BookService();

File: src/main/java/com/system/book/service/BookService.java
package com.system.book.service;

import com.system.book.dto.request.BookRequestDto;
import com.system.book.dto.response.BookResponseDto;
import com.system.book.model.Book;

import java.util.List;

public interface BookService {
    BookResponseDto saveBookInfo(BookRequestDto bookRequestDto);
    BookResponseDto getBookInfoById(Integer id);
    void deleteBookInfoById(Integer id);
    BookResponseDto updateBookInfo(Integer id, BookRequestDto bookRequestDto);
    List<BookResponseDto>getAllBook();
}
